plugin/src/main/java/com/denizenscript/denizen/tags/core/ServerTagBase.java

```diff
@@ -2341,34 +1782,38 @@ else if ((attribute.matches("plugins_handling_event") || attribute.matches("list
         // @example
         // - give <server.generate_loot_table[id=chests/spawn_bonus_chest;killer=<player>;location=<player.location>]>
         // -->
-        else if (attribute.startsWith("generate_loot_table") && attribute.hasParam()) {
-            MapTag map = attribute.inputParameterMap();
+        tagProcessor.registerTag(ListTag.class, MapTag.class, "generate_loot_table", (attribute, object, map) -> {
             ElementTag idObj = map.getRequiredObjectAs("id", ElementTag.class, attribute);
             LocationTag locationObj = map.getRequiredObjectAs("location", LocationTag.class, attribute);
             if (idObj == null || locationObj == null) {
-                return;
+                return null;
             }
-            NamespacedKey key = NamespacedKey.fromString(CoreUtilities.toLowerCase(idObj.toString()));
-            if (key == null) {
-                return;
+            LootTable table = Bukkit.getLootTable(Utilities.parseNamespacedKey(idObj.asLowerString()));
+            if (table == null) {
+                attribute.echoError("Invalid loot table ID '" + idObj + "' specified.");
+                return null;
             }
-            LootTable table = Bukkit.getLootTable(key);
             LootContext.Builder context = new LootContext.Builder(locationObj);
             EntityTag killer = map.getObjectAs("killer", EntityTag.class, attribute.context);
             ElementTag luck = map.getElement("luck");
             ElementTag bonus = map.getElement("loot_bonus");
             EntityTag entity = map.getObjectAs("entity", EntityTag.class, attribute.context);
             if (entity != null) {
-                context = context.lootedEntity(entity.getBukkitEntity());
+                context.lootedEntity(entity.getBukkitEntity());
             }
             if (killer != null) {
-                context = context.killer((HumanEntity) killer.getLivingEntity());
+                if (killer.getLivingEntity() instanceof HumanEntity humanEntity) {
+                    context.killer(humanEntity);
+                }
+                else {
+                    attribute.echoError("Invalid killer '" + killer + "' specified: must be an online player or a player-type NPC.");
+                }
             }
             if (luck != null) {
-                context = context.luck(luck.asElement().asFloat());
+                context.luck(luck.asFloat());

```

Comment:
`context = context` while functionally useless in practice is more proper to the design of the class being used (hypothetically it is valid for the method to be changed without warning such that it returns a copy with the value changed, rather than altering itself).

Reply:
I see, changed that since that's how builder classes usually behave, but fair enough - reverted